include/deal.II/base/tensor.h

```diff
@@ -103,6 +103,13 @@ class Tensor
   Tensor (const array_type &initializer);
 
   /**
+   * Copy constructor taking a tesnor of another base data type
+   */
+  template <typename OtherNumber>
+  explicit
+  Tensor (const Tensor<rank_,dim,OtherNumber> &);
+
```

Comment:
We often put a dummy name in the declaration to make things more similar to the implementation (but we are inconsequential about it, see below)